addons/crm/models/crm_lead.py

```diff
@@ -1907,6 +1955,7 @@ def _get_customer_information(self):
                 ).items() if val and key != 'email'  # don't force email used as criterion
             })
             values['is_company'] = is_company
+            values['parent_id'] = not is_company and lead.commercial_partner_id.id
```

Comment:
Smort

Reply:
Note, now under regular use it won't actually ever be `is_company`, since we don't fall back on partner_name anymore. And instead you'll always have either company_name or parent_id set, which imo makes more sense anyway.